UPGRADING.md

```diff
@@ -1,6 +1,12 @@
 # Upgrading from Psalm 4 to Psalm 5
 ## Changed
 
+- [BC] Psalm 5.1 will switch its internal representation of `list<T>` and `non-empty-list<T>` from the TList and TNonEmptyList classes to an unsealed list shape: the TList, TNonEmptyList and TCallableList classes will be removed.
```

Comment:
I'd suggest to follow semver and name that version 6.0

Reply:
I actually was hoping to get this merged by Psalm 5.0 given https://github.com/psalm/psalm.dev/commit/dbe30c025aa59ad602b1b3a1ab7792da021679e4, that description is outdated

Reply:
Probably not gonna make it though, I'd still love to get at least https://github.com/vimeo/psalm/pull/8728 merged so people already stop using TList and switch over to keyed lists in time for 5.1